### PR TITLE
Add unit test showing null Methodinvocation type

### PR DESCRIPTION
**Pull request by greg-dennis**
## What's your motivation?

Add unit test showing null Methodinvocation type. This test fails in this repo but the very same test passes when added to the `rewrite` repo.

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- [ ] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/test/java/org/openrewrite/java/testing/cleanup/KotlinTestMethodShouldBeUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            return method;
+        }
+    }
+
+    private static <T> T getOnly(Iterable<T> iterable) {
+        Iterator<T> iter = iterable.iterator();
+        T only = iter.next();
+        assertFalse(iter.hasNext());
+        return only;
+    }
```
Comment (Laurens-W) [inline src/test/java/org/openrewrite/java/testing/cleanup/KotlinTestMethodShouldBeUnitTest.java:78]:
```suggestion
    @Override
    public void defaults(RecipeSpec spec) {
        spec
          .parser(KotlinParser.builder());
    }

    @Test
    void showNullMethodInvocationType() {
        rewriteRun(
          //language=kotlin
          kotlin(
            """
              class ExampleTest {
                  fun myTest() = run {
                      return 1
                  }
              }
              """
          ));
    }
```
I've simplified it to this and am indeed seeing it fail in rewrite-testing-frameworks but succeed in rewrite

Comment (greg-dennis) [reply src/test/java/org/openrewrite/java/testing/cleanup/KotlinTestMethodShouldBeUnitTest.java:78]: Right, that simple example will show the symptom, but I found the failure to be more cryptic. The intention of the PR was to reveal the underlying issue: that the type of the `run` method invocation is null.

Comment (greg-dennis) [reply src/test/java/org/openrewrite/java/testing/cleanup/KotlinTestMethodShouldBeUnitTest.java:78]: @Laurens-W, there appears to be other code that "expects" the types of Kotlin method invocations to sometimes be null ([example](https://github.com/openrewrite/rewrite/blob/e118585d5a7081be8315c669127177e8b100002f/rewrite-kotlin/src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java#L2018)). If that is sometimes "expected" behavior, the following workaround might make sense: https://github.com/openrewrite/rewrite/pull/6409

Thoughts?

**Comment by timtebeek** [comment]
Wondering if we'll see improvement here following
- https://github.com/openrewrite/rewrite/pull/6766

If so we can close this PR.
